pkg/generator/generator.go

```diff
@@ -0,0 +1,110 @@
+package generator
+
+import (
+	"fmt"
+
+	"get.porter.sh/porter/pkg/secrets"
+	"github.com/cnabio/cnab-go/secrets/host"
+	"github.com/cnabio/cnab-go/valuesource"
+	survey "gopkg.in/AlecAivazis/survey.v1"
+)
+
+// GenerateOptions are the options to generate a parameter or credential set
+type GenerateOptions struct {
+	// Name of the parameter or credential set.
+	Name string
+
+	// Should we survey?
+	Silent bool
+}
+
+// SurveyType indicates whether the survey is for a parameter or credential
+type SurveyType string
+
+const (
+	surveyCredentials SurveyType = "credential"
+	surveyParameters  SurveyType = "parameter"
+
+	questionSecret  = "secret"
+	questionValue   = "specific value"
+	questionEnvVar  = "environment variable"
+	questionPath    = "file path"
+	questionCommand = "shell command"
+)
+
+type generator func(name string, surveyType SurveyType) (valuesource.Strategy, error)
+
+func genEmptySet(name string, surveyType SurveyType) (valuesource.Strategy, error) {
+	return valuesource.Strategy{
+		Name:   name,
+		Source: valuesource.Source{Value: "TODO"},
+	}, nil
+}
+
+func genSurvey(name string, surveyType SurveyType) (valuesource.Strategy, error) {
+	if surveyType != surveyCredentials && surveyType != surveyParameters {
+		return valuesource.Strategy{}, fmt.Errorf("unsupported survey type: %s", surveyType)
+	}
+
+	// extra space-suffix to align question and answer. Unfortunately misaligns help text
+	sourceTypePrompt := &survey.Select{
+		Message: fmt.Sprintf("How would you like to set %s %q\n ", surveyType, name),
+		Options: []string{questionSecret, questionValue, questionEnvVar, questionPath, questionCommand},
+		Default: "environment variable",
+	}
+
+	// extra space-suffix to align question and answer. Unfortunately misaligns help text
+	sourceValuePromptTemplate := "Enter the %s that will be used to set %s %q\n "
+
+	c := valuesource.Strategy{Name: name}
+
+	source := ""
+	if err := survey.AskOne(sourceTypePrompt, &source, nil); err != nil {
+		return c, err
+	}
+
+	// TODO: add 'skip' option *if* surveyType = surveyParameters *and* parameter not required
```

Comment:
Thoughts on these 2 TODOs?  Tackle in this PR?  Follow-ups?  (Assuming functionality is desired...)